### ISSUE TITLE
Sourcemap weirdness

### ISSUE THREAD

**Issue by ffoodd** [MohammadYounes/rtlcss#180]
Hi there!

I'm currently working on [implementing RTL on Bootstrap v5](https://github.com/twbs/bootstrap/pull/30980/), and while doing so I noticed [that sourcemaps for RTLCSS processed files have some weirdness](https://github.com/twbs/bootstrap/pull/30980/files?file-filters%5B%5D=.map#diff-03a3287204e2c4aec80ef1c2b200deab8c2ef99fdeef9676fb6dd4880275393e):
1. a new source is mentionned, not part of our sources and not a file either;
2. its path is URL encoded: `%3Cinput%20css%202%3E` (which results in `<input css 2>`);
3. and AFAIK seems to be mapping to the `/* rtl:raw */` directive.

So this is not really bug IMHO, however I think this should either be "customized" to mention something like "RTLCSS raw directive" or at least be mentionned somewhere in RTLCSS docs.

I'm not sure customizing sourcemaps in that case is doable, but would be really great :+1: 

Thanks for your invaluable work, I'd be pleased to have some review from you one day :heart:

**Comment by MohammadYounes** [comment]
Hi,

Thanks for such a Great PR, It's a joy to see rtlcss has finally made it to the official bootstrap 👍

Yes, indeed, parsing CSS strings without setting its input source produces such result. Do you think it's better to have it point to the original file instead of `<input ...>`   ?

**Comment by ffoodd** [comment]
If possible, I guess it'd be better indeed.

I'm wondering what you mean by "without setting its input source": is there some way to set the input on our side?

**Comment by MohammadYounes** [comment]
This need to be changed inside the [plugin](https://github.com/MohammadYounes/rtlcss/blob/master/lib/plugin.js#L52). The source can be set in postcss parse options 

```js
const root = postcss.parse(css, { from: 'a.css' })
root.source.input.from //=> "/home/ai/a.css"

const root = postcss.parse(css)
root.source.input.from //=> "<input css 1>"
```
see https://postcss.org/api/#input-from

**Comment by MohammadYounes** [comment]
@ffoodd  I made a quick fix, I would appreciate it if you update to [v2.6.2](https://www.npmjs.com/package/rtlcss/v/2.6.2) and let me know the result.

**Comment by ffoodd** [comment]
Nice, I need to check if it suits our need.

As I said, that's not a bug at all. Thanks for your help!

**Comment by ffoodd** [comment]
@MohammadYounes Thanks a lot, it's really better! We now have `bootstrap.rtl.css` as a source <3

**Comment by blimmer** [comment]
For folks who find this issue trying to debug their application, here's the fix that was released in 2.6.2: https://github.com/MohammadYounes/rtlcss/commit/80c15f218ffffcb66459953e60c65a0b27b89344
